api/types/src/transaction.rs

```diff
@@ -1260,6 +1314,13 @@ impl TryFrom<SingleKeySignature> for AccountAuthenticator {
                     .context("Failed to parse given public_key bytes as Secp256k1EcdsaSignature")?;
                 AnySignature::secp256k1_ecdsa(signature)
             },
+            Signature::WebAuthn(s) => {
+                let partial_authenticator_assertion_response = s
+                    .inner()
+                    .try_into()
+                    .context( "Failed to parse given public_key bytes as PartialAuthenticatorAssertionResponse")?;
```

Comment:
*signature* bytes, not *public_key* bytes!

Reply:
Haha i think you're right. I was using the same format as above

```rust
Signature::Ed25519(s) => {
                let signature = s
                    .inner()
                    .try_into()
                    .context("Failed to parse given public_key bytes as Ed25519Signature")?;
                AnySignature::ed25519(signature)
            },
            Signature::Secp256k1Ecdsa(s) => {
                let signature = s
                    .inner()
                    .try_into()
                    .context("Failed to parse given public_key bytes as Secp256k1EcdsaSignature")?;
                AnySignature::secp256k1_ecdsa(signature)
            },
```

I guess these are all wrong... I can go ahead and fix them

Reply:
I am not sure if it'll be backwards compatible to fix them... I was thinking of doing the same in another PR.

Reply:
hmmm... maybe @davidiw might know?

Reply:
I have the idea to fix the max

Reply:
Error messages like these, don't give you any ability to do automatic fixes.  In this event, oh the signature is wrong, which you can't do much about anyways.

Someone will _always_ do string comparison, and we can't help that, but the error codes are the only things that should be used by users (while logging string messages for human readability)